easybuild/framework/easyblock.py

```diff
@@ -4198,6 +4207,98 @@ def print_dry_run_note(loc, silent=True):
     dry_run_msg(msg, silent=silent)
 
 
+def persist_failed_compilation_log_and_artifacts(build_successful, application_log, silent, app, easyconfig):
+    persistent_dirs = {}
+
+    def get_dir(path, dirname):
```

Comment:
I still don't understand why you need this and cannot just create the new folder inside `initialize_persistent_dir_and_replicate_if_paths_distinct`. See below